recipes/json-schema-validator/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class JsonSchemaValidatorConan(ConanFile):
+    name = "json-schema-validator"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/pboettch/json-schema-validator"
+    description = "JSON schema validator for JSON for Modern C++ "
+    topics = ("json-schema-validator", "modern-json",
+              "schema-validation", "json")
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        min_vs_version = "16"
+        min_cppstd = "17" if self.settings.compiler == "Visual Studio" else "11"
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, min_cppstd)
+            min_vs_version = "15"
```

Comment:
Why visual studio is different here?

Reply:
because when you configure cppstd, it allows newer code to be compiled where it would have previously failed if the setting was not explicitly specified.

Reply:
so why don't you keep 15 only, as it support the entire c++17? changing the min version looks wrong to me

Reply:
Because when cppstd is not 17 it will not compile. Based on the profile settings the min version is manipulated/configured

Reply:
> so why don't you keep 15 only, as it support the entire c++17? changing the min version looks wrong to me

Because the default for 15 is C++14, you need to explicitly set `compiler.cppstd=17` in order to obtain those feature